go/DEVELOPER.md

```diff
@@ -0,0 +1,206 @@
+# Developer Guide
+
+This document describes how to set up your development environment to build and test the GLIDE for Redis Go wrapper.
+
+### Development Overview
+
+The GLIDE for Redis Go wrapper consists of both Go and Rust code. The Go and Rust components communicate in two ways:
+1. using the [protobuf](https://github.com/protocolbuffers/protobuf) protocol.
+2. using shared C objects. [cgo](https://pkg.go.dev/cmd/cgo) is used to interact with the C objects from Go code.
+
+### Build from source
+
+#### Prerequisites
+
+Software Dependencies
+
+-   Go
+-   GNU Make
+-   git
+-   GCC
+-   pkg-config
+-   protoc (protobuf compiler) >= v3.20.0
+-   openssl
+-   openssl-dev
+-   rustup
+
+
+**Dependencies installation for Ubuntu**
+
+```bash
+sudo apt update -y
+sudo apt install -y git gcc pkg-config openssl libssl-dev unzip make
+# Install Go
+sudo snap install go --classic
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for CentOS**
+
+```bash
+sudo yum update -y
+sudo yum install -y git gcc pkgconfig openssl openssl-devel unzip wget tar
+# Install Go
+wget https://go.dev/dl/go1.22.0.linux-amd64.tar.gz
+sudo tar -C /usr/local -xzf go1.22.0.linux-amd64.tar.gz
+export PATH="$PATH:/usr/local/go/bin"
+export PATH="$PATH:$HOME/go/bin"
+# Install rust
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+# Install protobuf compiler
+PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+curl -LO $PB_REL/download/v3.20.3/protoc-3.20.3-linux-x86_64.zip
+unzip protoc-3.20.3-linux-x86_64.zip -d $HOME/.local
+export PATH="$PATH:$HOME/.local/bin"
+# Check that the protobuf compiler is installed
+protoc --version
+```
+
+**Dependencies installation for MacOS**
+
+```bash
+brew update
+brew install go make git gcc pkgconfig protobuf@3 openssl
+export PATH="$PATH:$HOME/go/bin"
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+source "$HOME/.cargo/env"
+# Check that the Rust compiler is installed
+rustc --version
+```
+
+#### Building and installation steps
+
+Before starting this step, make sure you've installed all software requirements.
+
+1. Clone the repository:
+    ```bash
+    VERSION=0.1.0 # You can modify this to other released version or set it to "main" to get the unstable branch
+    git clone --branch ${VERSION} https://github.com/aws/glide-for-redis.git
+    cd glide-for-redis
+    ```
+2. Initialize git submodule:
```

Comment:
I'm not sure that it is worth including 1 & 2

Reply:
The node/python DEVELOPER.md includes them, so I suggest either matching that pattern here or changing all DEVELOPER.md files to remove them in a separate PR